src/game/physics.rs

```diff
@@ -0,0 +1,50 @@
+//! Run a very simple physics simulation.
+
+use bevy::{
+    ecs::component::{ComponentHooks, StorageType},
+    prelude::*,
+};
+
+pub(super) fn plugin(app: &mut App) {
+    // `FixedUpdate` runs before `Update`, so the physics simulation is advanced
+    // before the player's visual representation is updated.
+    app.add_systems(FixedUpdate, advance_physics);
+}
+
+/// How many units per second the player should move.
+#[derive(Debug, Component, Clone, Copy, PartialEq, Default, Deref, DerefMut)]
+pub(crate) struct Velocity(pub(crate) Vec3);
+
+/// The actual transform of the player in the physics simulation.
+/// This is separate from the `Transform`, which is merely a visual
+/// representation.
+#[derive(Debug, Clone, Copy, PartialEq, Default, Deref, DerefMut)]
+pub(crate) struct PhysicalTransform(pub(crate) Transform);
```

Comment:
I'm willing to believe this is needed but I'd really need to be convinced of it.

Reply:
This is needed because we cannot store 3 Transforms on a single entity without newtyping

Reply:
Well yeah, the question is moreso why do we need another Transform :smile:

Reply:
`Transform` is used for rendering, we need to store another one for FixedUpdate as well as a buffer for it's previous value

Reply:
See https://github.com/bevyengine/bevy/pull/14223 for a full writeup.